### DISCUSSION TITLE
Incorporating density-dependent porous flow for freshwater-seawater system

### DISCUSSION THREAD

**Discussion by jwheiss** [idaholab/moose#20334] [Q&A General] [answered]
Hi all,

I would like to incorporate density-dependent flow in PorousFlow to simulate flow in a coastal aquifer where seawater and freshwater mix (1 solute (total salt) affects fluid density). I want to use a simple linear relationship between fluid density and salt concentration: rho_fluid = rho_fresh + (delta_rho/delta_C) * C
where C is the salt mass of the fluid (max of 35 g/kg for seawater) and delta_rho/delta_C = 0.71. 

My current model is set up as single fluid density coupled to geomechanics. I attempted to use the example 1phase_convection.i model to inform how to do this, but the example model is more complex with multiple phases and I am having difficulty following the contents of the 1phase_convection.i file since it does not contain line by line comments.  The 2phase_convection model at https://mooseframework.inl.gov/modules/porous_flow/lava_lamp.html has been helpful to a degree - I am looking for help to incorporate a simpler density:concentration relationship in a model with one solute. Has anyone modeled density-dependent flow in a saltwater-freshwater system, or have suggestions for how to incorporate it?

Thanks,
James

**Comment by cpgr** [comment, marked as answer]
I don't think that there is anything exactly like you want already in PorousFlow, but it wouldn't be too hard to make a new material that calculates the density according to your requirements. 

I have done something similar to this in the past (specifically for density-driven convective mixing) that predates PorousFlow. You can take a look at what I did here https://github.com/cpgr/numbat - from memory the density is computed just like you want.

**Reply by jwheiss** [reply]
Hi, thanks for that link. The 2D.i numbat example problem specifies fluid density in the Materials block as

  [./density]
    type = NumbatDensity
    concentration = concentration
    zero_density = 995
    delta_density = 10.5
    saturated_concentration = 0.049306
  [../]

If I create a new Object in MOOSE equivalent to the NumbatDensity Object in numbat, it should do the trick. 

Thanks,
James

**Reply by cpgr** [reply]
Let us know if you run into any issues.
